docs/examples/pvfleets-qa-pipeline/pvfleets-temperature-qa.py

```diff
@@ -0,0 +1,242 @@
+    time_series, limits=(-40, 185))
+temperature_limit_mask = temperature_limit_mask.reindex(
+    index=time_series.index,
+    method='ffill',
+    fill_value=False)
+
+# FIND OUTLIERS (Z-SCORE FILTER)
+zscore_outlier_mask = zscore(time_series,
+                             zmax=4,
+                             nan_policy='omit')
+
+# PERFORM ADDITIONAL CHECKS, INCLUDING CHECKING UNITS (CELSIUS OR FAHRENHEIT)
+temperature_mean = time_series.mean()
+if temperature_mean > 35:
+    temp_units = 'F'
+else:
+    temp_units = 'C'
+
+print("Estimated Temperature units: " + str(temp_units))
+
+# Run additional checks based on temperature sensor type.
+if data_stream_type == 'module':
+    if temp_units == 'C':
+        module_limit_mask = (time_series <= 85)
+        temperature_limit_mask = (temperature_limit_mask & module_limit_mask)
+if data_stream_type == 'ambient':
+    ambient_limit_mask = pvanalytics.quality.weather.temperature_limits(
+        time_series, limits=(-40, 120))
+    temperature_limit_mask = (temperature_limit_mask & ambient_limit_mask)
+    if temp_units == 'C':
+        ambient_limit_mask_2 = (time_series <= 50)
+        temperature_limit_mask = (temperature_limit_mask &
+                                  ambient_limit_mask_2)
+# Get the percentage of data flagged for each issue, so it can later be logged
+pct_stale = round((len(time_series[
+    stale_data_mask].dropna())/len(time_series.dropna())*100), 1)
+pct_erroneous = round((len(time_series[
+    ~temperature_limit_mask].dropna())/len(time_series.dropna())*100), 1)
+pct_outlier = round((len(time_series[
+    zscore_outlier_mask].dropna())/len(time_series.dropna())*100), 1)
+
+# Visualize all of the time series issues (stale, abnormal, outlier)
+time_series.plot()
+labels = ["Temperature"]
+if any(stale_data_mask):
+    time_series.loc[stale_data_mask].plot(ls='',
+                                          marker='o',
+                                          color="green")
+    labels.append("Stale")
+if any(~temperature_limit_mask):
+    time_series.loc[~temperature_limit_mask].plot(ls='',
+                                                  marker='o',
+                                                  color="yellow")
+    labels.append("Abnormal")
+if any(zscore_outlier_mask):
+    time_series.loc[zscore_outlier_mask].plot(ls='',
+                                              marker='o',
+                                              color="purple")
+    labels.append("Outlier")
+plt.legend(labels=labels)
+plt.title("Time Series Labeled for Basic Issues")
+plt.xticks(rotation=20)
+plt.xlabel("Date")
+plt.ylabel("Temperature")
+plt.tight_layout()
+plt.show()
+
+# %%
+# Now, let's filter out any of the flagged data from the basic temperature
+# checks (stale or abnormal data). Then we can re-visualize the data
+# post-filtering.
+
+# Filter the time series, taking out all of the issues
+time_series = time_series[~stale_data_mask]
```

Comment:
Same comment as in the irradiance example: do these filters change the length of the `Series`?

Reply:
Did the same here as above to make the mask a single call. asfreq() returns the time series to original data frequency with NaN's:

```python
# Filter the time series, taking out all of the issues
issue_mask = ((~stale_data_mask) & (temperature_limit_mask) &
              (~zscore_outlier_mask))
time_series = time_series[issue_mask]
time_series = time_series.asfreq(data_freq)
```